enterprise/cmd/frontend/internal/auth/oauth/middleware.go

```diff
@@ -69,12 +69,17 @@ func newOAuthFlowHandler(db database.DB, serviceType string) http.Handler {
 			http.Error(w, "Misconfigured GitHub auth provider.", http.StatusInternalServerError)
 			return
 		}
-
-		extraScopes, err := getExtraScopes(req.Context(), db, serviceType)
-		if err != nil {
-			log15.Error("Getting extra OAuth scopes", "error", err)
-			http.Error(w, "Authentication failed. Try signing in again (and clearing cookies for the current site).", http.StatusInternalServerError)
-			return
+		op := req.URL.Query().Get("op")
+		extraScopes := []string{}
+		var err error
+		if op == string(LoginStateOpCreateCodeHostConnection) {
```

Comment:
nit: This looks more strongly typed but after all changes nothing in this particular case 😁 

```suggestion
		op := LoginStateOp(req.URL.Query().Get("op"))
		extraScopes := []string{}
		var err error
		if op == LoginStateOpCreateCodeHostConnection {
```

Reply:
TIL, I assumed "conversion" to LoginStateOp will fail at runtime - guess that's what would happen in Typescript https://www.typescriptlang.org/play?pretty=true#code/FDAuE8AcFMAIBkD2BzAlgOwMqgIamgPKSwC8sAROQD7kDGATtHtAMKIAm0AEogM6ht06aLVCpE6cgG5gAG2ihYslBgBcSNFlz4ipCuVgz5i2hOGj1Krc11k6jZm049+g82IkGjC2ACMc7LCWmtg2kHb+7F4gpui8iPIAdMrIABQpGAA0se6ZkQCUQA

Reply:
I need to be really careful about different behaviors when I learn TS 😁